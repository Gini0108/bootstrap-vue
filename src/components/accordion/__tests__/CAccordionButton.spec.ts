import { mount } from '@vue/test-utils'
import { CAccordionButton as Component } from '../../../index'

const ComponentName = 'CAccordionButton'

const defaultWrapper = mount(Component, {
  propsData: {
    collapsed: true
  },
  slots: {
    default: 'Default slot',
  }
})

describe(`Loads and display ${ComponentName} component`, () => {
  it('has a name', () => {
    expect(Component.name).toMatch(ComponentName)
  })
  it('renders correctly', () => {
    expect(defaultWrapper.html()).toMatchSnapshot()
  }) 
  it('contain slots and classes', () => {
    expect(defaultWrapper.text()).toContain('Default slot')
    expect(defaultWrapper.classes('accordion-button')).toBe(true)
    expect(defaultWrapper.classes('collapsed')).toBe(true)
  })
})