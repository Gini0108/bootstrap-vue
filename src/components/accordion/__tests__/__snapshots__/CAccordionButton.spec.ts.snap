// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loads and display CAccordionButton component renders correctly 1`] = `"<button class=\\"accordion-button collapsed\\">Default slot</button>"`;
